src/style-spec/feature_filter/index.js

```diff
@@ -74,21 +75,174 @@ const filterSpec = {
  */
 function createFilter(filter: any): FeatureFilter {
     if (filter === null || filter === undefined) {
-        return {filter: () => true, needGeometry: false};
+        return {filter: () => true, dynamicFilter: () => true, needGeometry: false};
     }
 
     if (!isExpressionFilter(filter)) {
         filter = convertFilter(filter);
     }
 
-    const compiled = createExpression(filter, filterSpec);
-    if (compiled.result === 'error') {
-        throw new Error(compiled.value.map(err => `${err.key}: ${err.message}`).join(', '));
+    let staticFilter = true;
+    try {
+        staticFilter = extractStaticFilter(filter);
+    } catch (e) {
+        console.warn(`Failed to extract static filter from ${JSON.stringify(filter)}. Falling back to using 'true' as a filter and evaluating entire filter dynamically`);
```

Comment:
This warning doesn't seem super helpful for end users. Is there a more concrete instruction that we could provide?

Reply:
I updated it, added more formatting and a link to our bug report page.
This is what it looks like
<img width="1328" alt="Screen Shot 2021-09-14 at 12 28 40 PM" src="https://user-images.githubusercontent.com/1449257/133322079-ef9c6c48-45af-4015-a7ef-f5eebe87cd0b.png">